docs/development.md

```diff
@@ -0,0 +1,81 @@
+# Local development
+
+To do proper development and testing of the GitOps Service there are some requirements you need to set up first.
+
+**Development Environment Requirements**:
+
+1. GNU/Linux or Mac Operating System (Windows with WSL/WSL2 or MSYS2, should also work, but is beyond the scope of this document.)
+2. Running a local Kubernetes cluster: Our team uses [KinD] or [K3s] for simplicity.
+You can use [kubectx/kubens](https://github.com/ahmetb/kubectx) to manage Kubernetes context if you like.
+3. A container registry that you and your Kubernetes cluster can reach. We recommend [quay.io](https://quay.io/signin/).
+4. User-permissions set so scripts can start/stop containers and interact with [Docker].
+5. There are other binaries (`kubectl`, `helm`, _etc_) required as well, which the Makefile will prompt you to install them when it needs them. Install the binary dependencies: `make download-deps`
+
+Once you have a test environment already setup, you are ready to proceed with either the Docker or the Kubernetes workflow.
+
+**Option 1: Development Workflow with Docker**:
+
+Currently, the project has a mixture of unit-tests and integration tests, meaning running `go test` for some components, it is going to fail _unless_ you have the required test environment up and running.
+
+To do that, follow these steps:
+
+1. Have a local (or remote) Kubernetes cluster up and running (_e.g._ `kind create cluster`)
+2. Install the namespaces, CRDs and RBAC resources for both `cluster-agent` and `backend` and start the local containers (PostgreSQL and its UI dashboard): `make devenv-docker`.
+3. Start `goreman` that starts all the required processes in the same terminal: `make start`
+
+Now, you can run the tests:
+
+```shell
+make test-backend-shared            # Run test for backend-shared only
+make test-backend                   # Run tests for backend only
+make test-cluster-agent             # Run test for cluster-agent only
+make test                           # Or, run tests for all components
+```
+
+To reset the database run, `make reset-db`.
+
+**Option 2: Development Workflow with Kubernetes**:
```

Comment:
We need to be encouraging contributors to develop on OpenShift directly to ensure dev env is as close to the production env as possible.

Reply:
We shouldn't care about Kubernetes/kind at all.

Reply:
See my private comments on Slack for this :smile:.